
package com.mycalc;

import javax.jws.WebMethod;
import javax.jws.WebParam;
import javax.jws.WebResult;
import javax.jws.WebService;
import javax.xml.bind.annotation.XmlSeeAlso;
import javax.xml.ws.Action;
import javax.xml.ws.FaultAction;
import javax.xml.ws.RequestWrapper;
import javax.xml.ws.ResponseWrapper;


/**
 * This class was generated by the JAX-WS RI.
 * JAX-WS RI 2.2.4-b01
 * Generated source version: 2.2
 * 
 */
@WebService(name = "MyName", targetNamespace = "MyCalc.com")
@XmlSeeAlso({
    ObjectFactory.class
})
public interface MyName {


    /**
     * 
     * @param arg1
     * @param arg0
     * @return
     *     returns int
     */
    @WebMethod
    @WebResult(targetNamespace = "")
    @RequestWrapper(localName = "addition", targetNamespace = "MyCalc.com", className = "com.mycalc.Addition")
    @ResponseWrapper(localName = "additionResponse", targetNamespace = "MyCalc.com", className = "com.mycalc.AdditionResponse")
    @Action(input = "MyCalc.com/MyName/additionRequest", output = "MyCalc.com/MyName/additionResponse")
    public int addition(
        @WebParam(name = "arg0", targetNamespace = "")
        int arg0,
        @WebParam(name = "arg1", targetNamespace = "")
        int arg1);

    /**
     * 
     * @param arg1
     * @param arg0
     * @return
     *     returns int
     * @throws MyException_Exception
     */
    @WebMethod
    @WebResult(targetNamespace = "")
    @RequestWrapper(localName = "division", targetNamespace = "MyCalc.com", className = "com.mycalc.Division")
    @ResponseWrapper(localName = "divisionResponse", targetNamespace = "MyCalc.com", className = "com.mycalc.DivisionResponse")
    @Action(input = "MyCalc.com/MyName/divisionRequest", output = "MyCalc.com/MyName/divisionResponse", fault = {
        @FaultAction(className = MyException_Exception.class, value = "MyCalc.com/MyName/division/Fault/MyException")
    })
    public int division(
        @WebParam(name = "arg0", targetNamespace = "")
        String arg0,
        @WebParam(name = "arg1", targetNamespace = "")
        String arg1)
        throws MyException_Exception
    ;

}
